daq/stream_monitor.py

```diff
@@ -16,23 +19,24 @@ def __init__(self, timeout_ms=None, idle_handler=None, loop_hook=None):
         self.loop_hook = loop_hook
         self.poller = select.poll()
         self.callbacks = {}
-        self.dirty = False
 
     def get_fd(self, target):
         """Return the fd from a stream object, or fd directly"""
         return target.fileno() if 'fileno' in dir(target) else target
 
     # pylint: disable=too-many-arguments
-    def monitor(self, name, desc, callback=None, hangup=None, copy_to=None, error=None):
+    def monitor(self, name, desc, callback=None, hangup=None, copy_to=None,
+                error=None, timeout_sec=120):
```

Comment:
better to leave None as the default for timeout.  "120" is a very arbitrary constant and really has no meaning within the scope of this function.  If it really does make sense to have a default timeout, then it should be passed into the constructor of stream_monitor itself.